pkg/container-utils/testutils/k8s.go

```diff
@@ -108,9 +96,37 @@ spec:
   terminationGracePeriodSeconds: 0
   containers:
   - name: %s
-    image: %s%s%s
+    image: %s
+`, podname, namespace, podname, podname, image)
+
+	if cmd != "" {
+		cmdLine := fmt.Sprintf("    command: %s\n", cmd)
+
+		commandArgsLine := ""
+		if commandArgs != "" {
+			commandArgsLine = fmt.Sprintf("    args:\n    - %s\n", commandArgs)
+		}
+		yamlStr = yamlStr + cmdLine + commandArgsLine
+	}
+
+	if len(limits) > 0 {
+		yamlStr = yamlStr + "    resources:\n      limits:\n"
+		for k, v := range limits {
+			yamlStr = yamlStr + fmt.Sprintf("        %s: \"%s\"\n", k, v)
+		}
+	}
+
+	return yamlStr
```

Comment:
The best option would probably be to use the API instead of calling `kubectl apply ...` but that is what we had triet some time ago and saw that there is too much overhead: #1116